sync/sync_head.go

```diff
@@ -130,6 +131,9 @@ func (s *Syncer[H]) setSubjectiveHead(ctx context.Context, netHead H) {
 	s.pending.Add(netHead)
 	s.wantSync()
 	log.Infow("new network head", "height", netHead.Height(), "hash", netHead.Hash())
+	s.metrics.observeNewHead(int64(netHead.Height()))
+	s.metrics.observeHeaderTimestamp(netHead.Time())
+	s.metrics.observeLaggingHeader()
```

Comment:
Looks like those are correlated and called at the same time. Might be worth making one function instead of 3 then.

Reply:
reworked